workflow-templates/upgrade-python-requirements.yml

```diff
@@ -0,0 +1,52 @@
+name: Upgrade Requirements
+
+on:
+  schedule:
+    # will start the job at midnight every monday (UTC), should be updated to use appropriate value
+    - cron: "0 0 * * 1"
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: 'master'
+
+jobs:
+  upgrade_requirements:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: make upgrade
+        run: |
+          cd $GITHUB_WORKSPACE
+          make upgrade
+
+      - name: setup target branch
+        run: echo "target_branch=$(if ['${{ github.event.inputs.branch }}' = '']; then echo 'master'; else echo '${{ github.event.inputs.branch }}'; fi)" >> $GITHUB_ENV
+
+      - name: setup testeng-ci
+        run: |
+          git clone https://github.com/edx/testeng-ci.git
+          cd $GITHUB_WORKSPACE/testeng-ci
+          pip install -r requirements/base.txt
+      - name: create pull request
+        env:
+          GITHUB_TOKEN: ${{ secrets.REQUIREMENTS_BOT_GITHUB_TOKEN }}
+          GITHUB_USER_EMAIL: ${{ secrets.REQUIREMENTS_BOT_GITHUB_EMAIL }}
+        run: |  # replace user-reviewers and team-reviewers accordingly
+          cd $GITHUB_WORKSPACE/testeng-ci
+          python -m jenkins.pull_request_creator --repo-root=$GITHUB_WORKSPACE \
```

Comment:
[passing thought] This is not relevant to the PR as is, but it might be useful to move this script to a repo that we ship to PyPI so that we can get the benefit of just pip installing what we need instead of needing to clone another repo.

Reply:
yes, we can definitely get this published to PyPI, we've had a similar discussion when we published `repo-tools` to PyPI, what do you think about publishing _testeng-ci_ @jmbowman @nedbat ?

Reply:
If we're already shipping repo-tools, it might make sense to just move this script to that repo also.